### PR TITLE
Add basic documentation for sub searches

### PR DESCRIPTION
**Pull request by jdconrad**
This adds basic documentation for the `sub_searches` top-level element in the search API. (https://github.com/elastic/elasticsearch/pull/96224).

### CODE REVIEW

File: docs/reference/search/rrf.asciidoc [hunk 1]
```diff
@@ -110,9 +112,58 @@ RRF does not currently support:
...
+                }
+            }
+        },
+        {
+            "query": {
+                "term": {
+                    "text": "red shoes"
+                }
+            }
+        }
```
Comment (javanna) [inline docs/reference/search/rrf.asciidoc:144]: what is the end goal of this request? Should we provide an example that is closer to how we expect users to use this feature?

Comment (jdconrad) [reply docs/reference/search/rrf.asciidoc:144]: Yes. I will try re-work this with an ELSER query instead.

Comment (benwtrent) [reply docs/reference/search/rrf.asciidoc:144]: I mentioned the same thing in my review, but just noticed it was still `pending` and I didn't hit submit 🤦 .

The short of it, I agree with @javanna , we should use `text_expansion` or `script_query`. Something that scores in a different distribution than BM25

File: docs/reference/search/rrf.asciidoc [hunk 2]
```diff
@@ -32,10 +32,11 @@ return score
...
 
 You can use RRF as part of a <<search-search, search>> to combine and rank
-documents using multiple result sets from
-
-* 1 query and 1 or more kNN searches
-* 2 or more kNN searches
+documents using result sets from a combination of
+<<request-body-search-query, query>>,
+<<request-body-sub-searches, sub searches>>, and/or
+<<search-api-knn, knn searches>>. A minimum of 2 results sets for
```
Comment (mayya-sharipova) [inline docs/reference/search/rrf.asciidoc:38]: should we drop the last word "for" here?

File: docs/reference/search/rrf.asciidoc [hunk 3]
```diff
@@ -110,9 +112,58 @@ RRF does not currently support:
 * <<request-body-search-explain, explain>>
 * <<profiling-queries, profiling>>
 
-Using unsupported features as part of a search using RRF will result
+Using unsupported features as part of a search using RRF results
```
Comment (mayya-sharipova) [inline docs/reference/search/rrf.asciidoc:115]: "a search using RRF" => "a search with RRF" as there are two "using" in this sentence?

File: docs/reference/search/rrf.asciidoc (hunk 2 above)
Comment (benwtrent) [reply docs/reference/search/rrf.asciidoc:38]: you are correct .`for` should be removed

Comment (jdconrad) [reply docs/reference/search/rrf.asciidoc:38]: Good catch! Fixed.

File: docs/reference/search/rrf.asciidoc (hunk 3 above)
Comment (jdconrad) [reply docs/reference/search/rrf.asciidoc:115]: Changed.

File: docs/reference/search/rrf.asciidoc (hunk 1 above)
Comment (jdconrad) [reply docs/reference/search/rrf.asciidoc:144]: Updated to use a text expansion query.

File: docs/reference/search/rrf.asciidoc [hunk 4]
```diff
@@ -110,9 +112,58 @@ RRF does not currently support:
...
+In the above example, we execute each of the two sub searches
+independently of each other. First we run the term query for
+`blue shoes`, and then we run the term query for `red shoes`.
+Afterwards, we use RRF to combine the results on a coordinating node.
+While this example can be replaced with a simple boolean query
+because both sub searches use the BM25 scoring algorithm, sub searches
+ranked with RRF becomes extremely useful when each sub search uses a
+different scoring algorithm. A BM25 sub search combined with an
+<<semantic-search-elser, ELSER>> sub search can give relevant results
+significantly better than one or the other used on their own.
```
[the remark this answers is not in the document]
Comment (jdconrad) [reply docs/reference/search/rrf.asciidoc:165]: Updated to use a text expansion query.

**Comment by jdconrad** [comment]
@benwtrent @mayya-sharipova @javanna Thank you all for the reviews. I've update this based on your feedback.

**Review by benwtrent — APPROVED** [review]

**Comment by leemthompo** [comment]
@jdconrad

It would be nice to see an example response here and a small explanation of how to read that response and understand the value of using RRF. I'm working on a notebook that will use rrf with `sub_searches`, and it's necessary to give at least a minimal indication of the comparative advantage of this approach.

Happy to help copyedit, pair, or even turn rough notes into docs-ready section if anyone's game! 😄

**Pull request merged**